core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1345,7 +1354,7 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
 
             List<Field> outputFields = fields.build();
 
-            analyzeFiltersAndMasks(table, name, tableHandle, outputFields, session.getIdentity().getUser());
+            analyzeFiltersAndMasks(table, targetTableName, tableHandle, outputFields, session.getIdentity().getUser());
```

Comment:
Let's make `analyzeFiltersAndMasks` handle redirections itself

Reply:
I think we need to handle at a higher level, because we are not redirecting views, and `createScopeForView` also uses this method.